test/jdk/java/io/File/TempDirDoesNotExist.java

```diff
@@ -25,91 +25,154 @@
+                                      new String[] {
+                                          "TempDirDoesNotExist",
+                                          "io"
+                                      }),
+                         Arguments.of(0, WARNING,
+                                      new String[] {
+                                          "TempDirDoesNotExist", "nio"
+                                      }),
+                         Arguments.of(0, WARNING,
+                                      new String[] {
+                                          "-Djava.io.tmpdir=" + USER_DIR,
+                                          "TempDirDoesNotExist", "io"
+                                      }),
+                         Arguments.of(0, WARNING,
+                                      new String[] {
+                                          "-Djava.io.tmpdir=" + USER_DIR,
+                                          "TempDirDoesNotExist", "nio"
+                                      }));
+    }
 
-            // error message printed only once
-            testMessageCounter(0, "-Djava.io.tmpdir=" + tempDir,
-                    "TempDirDoesNotExist", "io", "nio");
-        }
+    public static Stream<Arguments> counterSource() {
+        // standard test with default setting for java.io.tmpdir
+        return Stream.of(Arguments.of(0,
+                                      new String[] {
+                                          "-Djava.io.tmpdir=" + tempDir(),
+                                          "TempDirDoesNotExist", "io", "nio"
+                                      }));
     }
 
-    private static void testMessageExist(int exitValue, String errorMsg, String... options) throws Exception {
-        ProcessTools.executeTestJvm(options).shouldContain(errorMsg)
-                .shouldHaveExitValue(exitValue);
+    @ParameterizedTest
+    @MethodSource("tempDirSource")
+    public void existingMessage(int exitValue, String errorMsg,
+                                String... options) throws Exception {
+       ProcessTools.executeTestJvm(options).shouldContain(errorMsg)
+           .shouldHaveExitValue(exitValue);
     }
 
-    private static void testMessageNotExist(int exitValue, String errorMsg,String... options) throws Exception {
-        ProcessTools.executeTestJvm(options).shouldNotContain(errorMsg).shouldHaveExitValue(exitValue);
+    @ParameterizedTest
+    @MethodSource("noTempDirSource")
+    public void nonexistentMessage(int exitValue, String errorMsg,
+                                   String... options) throws Exception {
+        ProcessTools.executeTestJvm(options).shouldNotContain(errorMsg)
+            .shouldHaveExitValue(exitValue);
     }
 
-    private static void testMessageCounter(int exitValue,String... options) throws Exception {
+    @ParameterizedTest
+    @MethodSource("counterSource")
+    public  void messageCounter(int exitValue, String... options)
+        throws Exception {
```

Comment:
```suggestion
    public void messageCounter(int exitValue, String... options)
            throws Exception {
```

Reply:
Thanks; will fix.

Reply:
Fixed in b3ac8c7b1242281bda86f7dc98efccf2bcefc7f6.